ERCS/erc-7580.md

```diff
@@ -0,0 +1,88 @@
+3. after the target promotion action ends, project contracts called the inferface `onTrackEnd(id, contract_address, function_hash)`
+4. promotion contract collect the project action info and distribute the rewards back to projects
+
+For example, we have two entities holding their respective contracts: contract A and contract B. Contract A targets on users who did specific key moves(eg. commit specific functions) in contract B and would give bonus/airdrop to these users. Sure B would also get incentives in the meanwhile. To connect all these dots, B needs to identity these users, verify they're coming for the A's bonus. Hence, we need a track mechanism to facilitate such business.
+
+## Specification
+
+The keywords “MUST,” “MUST NOT,” “REQUIRED,” “SHALL,” “SHALL NOT,” “SHOULD,” “SHOULD NOT,” “RECOMMENDED,” “MAY,” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Interfaces
+
+This protocol standardizes how to keep tracks on inter dapps operations, which initially offers 2 main methos onTrackStart and onTrackEnd.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.10;
+
+interface IERCXXX {
+    // Events
+
+    /// @dev Emits when track starts.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function with params
+    event onTrackStartRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    /// @dev Emits when track starts.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function with params
+    event onTrackEndRecorded(uint256 track_id, address contract_address, bytes32 function_hash);
+
+    // Functions
+
+    /// @dev Track a specified contract function start move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function with params
+    function onTrackStart(uint256 track_id, address contract_address, bytes32 function_hash) external;
+
+    /// @dev Track a specified contract function end move.
+    /// @param track_id track id
+    /// @param contract_address the address of tracking contract
+    /// @param function_hash the hash of tracking function with params
+    function onTrackEnd(uint256 track_id, address contract_address, bytes32 function_hash);
+}
+```
+
+
+## Rationale
+
+The core mechanism for this proposal is to provide a shared track interface standard for inter dapps, to improve the efficiency and fulfile the required tracking business. We provide two interface functions onTrackStart and onTrackEnd to fill the basic required info and connect the necessary dots. Sure there're more demands for more functions and it would be updated later.
```

Comment:
```suggestion
The core mechanism for this proposal is to provide a shared tracking interface for inter-dapp operations, to improve the efficiency and fulfill the required tracking business. We provide two interface functions `onTrackStart` and `onTrackEnd` to fill the basic required info and connect the necessary dots. Sure there're more demands for more functions and it would be updated later.
```